src/language-js/comments.js

```diff
@@ -807,32 +780,35 @@ function handleImportDeclarationComments({
 }
 
 function handleAssignmentPatternComments({ comment, enclosingNode }) {
-  if (enclosingNode && enclosingNode.type === "AssignmentPattern") {
+  if (enclosingNode?.type === "AssignmentPattern") {
     addLeadingComment(enclosingNode, comment);
     return true;
   }
   return false;
 }
 
+const assignmentLikeNodeTypes = new Set([
+  "VariableDeclarator",
+  "AssignmentExpression",
+  "TypeAlias",
+  "TSTypeAliasDeclaration",
+]);
+const complexExprNodeTypes = new Set([
+  "ObjectExpression",
+  "ArrayExpression",
+  "TemplateLiteral",
+  "TaggedTemplateExpression",
+  "ObjectTypeAnnotation",
+  "TSTypeLiteral",
+]);
 function handleVariableDeclaratorComments({
   comment,
   enclosingNode,
   followingNode,
 }) {
   if (
-    enclosingNode &&
-    (enclosingNode.type === "VariableDeclarator" ||
-      enclosingNode.type === "AssignmentExpression" ||
-      enclosingNode.type === "TypeAlias" ||
-      enclosingNode.type === "TSTypeAliasDeclaration") &&
-    followingNode &&
-    (followingNode.type === "ObjectExpression" ||
-      followingNode.type === "ArrayExpression" ||
-      followingNode.type === "TemplateLiteral" ||
-      followingNode.type === "TaggedTemplateExpression" ||
-      followingNode.type === "ObjectTypeAnnotation" ||
-      followingNode.type === "TSTypeLiteral" ||
-      isBlockComment(comment))
+    assignmentLikeNodeTypes.has(enclosingNode?.type) &&
+    (complexExprNodeTypes.has(followingNode?.type) || isBlockComment(comment))
```

Comment:
This changed a little bit. Previous one require `followingNode` not falsely, your version don't.

Reply:
[4c5e28f](https://github.com/prettier/prettier/pull/12747/commits/4c5e28f2a1c2a683afee3dff738ea5e123e565cb)